qiling/arch/utils.py

```diff
@@ -53,6 +53,11 @@ def disassembler(self, ql: Qiling, address: int, size: int):
 
         for insn in ql.arch.disassembler.disasm(data, address):
             offset = insn.address - ba
+            try:
+                # use r2 to get fine-grained name and offset
+                name, offset = ql.r2.resolve(insn.address)
```

Comment:
As mentioned elsewhere, r2 should not be an integral part of core, but assuming that it is - this is an ineffective way to do this. Better rely on the r2 enable flag rather than set an excpetion frame.

Reply:
"this is an ineffective way to do this" is not correct. Note r2 flags have more information than we have for now. For example, Qiling can't retrieve cross reference by any means for now.

Reply:
I think "ineffective" refers to the `try ... except` block. Anyway, if `r2` can be integrated into qiling, I think it's better to be a `property` so we can construct `R2` at need without specifying it at first.